omod/src/main/java/org/openmrs/module/providermanagement/fragment/controller/ProviderEditFragmentController.java

```diff
@@ -312,6 +313,43 @@ else if (ProviderManagementUtils.clearTimeComponent(date).after(new Date())) {
 
     }
 
+    public FragmentActionResult editSupervisor(@RequestParam(value = "supervisee", required = true) Person supervisee,
+                                               @RequestParam(value = "supervisor", required = true) Person supervisor,
+                                               @RequestParam(value = "relationship", required = false) Relationship relationship,
+                                               @RequestParam(value = "date", required = false) Date date,
+                                               @SpringBean("providerManagementService") ProviderManagementService providerManagementService) {
+
+        if (supervisee == null) {
+            return new FailureResult(Context.getMessageSourceService().getMessage("providermanagement.errors.supervisee.required"));
+        }
+
+        if (date == null) {
+            date = new Date();
+        }
+        if (relationship != null) {
```

Comment:
So, in this case, the relationship is the existing relationship supervisor-supervisee relationship, and if the new supervisor is differnent it ends the current relationship? (Makes sense, just wanted to make sure I followed it properly).